lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -606,6 +646,18 @@ fragmentId inps outs wits =
         putWord8 (messageTypeTag MsgTypeTransaction)
         putSignedTx inps outs wits
 
+delegationFragmentId
+    :: PoolId
+    -> ByteString
+    -> [(TxIn, Coin)]
+    -> [TxOut]
+    -> [TxWitness]
+    -> Hash "Tx"
+delegationFragmentId poolId accId inps outs wits =
```

Comment:
here, my assumption is that the id of stake delegation tx is computed as below

Reply:
See "FRAGMENT-ID" in https://github.com/input-output-hk/chain-libs/blob/7a7c74359df7b7fbdb9f71a83c11f88100897c06/chain-impl-mockchain/doc/format.md#block-content

Reply:
Although, @Anviking's approach to compute the fragment id directly in the `getMessage` decoder should also work here.

Reply:
>  @Anviking's approach to compute the fragment id directly in the getMessage decoder should also work here.

🤔 I'm not sure it works… @paweljakubas uses `delegationFragmentId ` here to to pass a valid txId when constructing the `Tx` in `StakeDelegationTx`.

Needing to embed the fragmentId inside the fragment itself is… inconvenient. Not sure if things (like already having `Tx` in `Message`) are subtly very off here or not 🤔 🤷‍♂